packages/connect-core/src/types.ts

```diff
@@ -1,15 +1,140 @@
 import { providers as ethersProviders, utils as ethersUtils } from 'ethers'
 import { Address, Network } from '@aragon/connect-types'
+
 import IOrganizationConnector from './connections/IOrganizationConnector'
+import App from './entities/App'
+import Transaction from './entities/Transaction'
 
-export type Metadata = (AragonArtifact | AragonManifest)[]
+export type Abi = (ethersUtils.EventFragment | ethersUtils.FunctionFragment)[]
+
+export type ConnectionContext = {
+  actAs: Address | null
+  ethereumProvider: object | null
+  ethersProvider: ethersProviders.Provider
+  ipfs: (cid: string) => string
+  network: Network
+  orgAddress: Address
+  orgConnector: IOrganizationConnector
+  orgLocation: string
+  verbose: boolean
+}
 
-// Type definition: https://github.com/ethers-io/ethers.js/blob/ethers-v5-beta/packages/abi/lib/fragments.d.ts#L68
-export type FunctionFragment = ethersUtils.FunctionFragment
+export type AppOrAddress = App | Address
 
-export type Abi = (ethersUtils.EventFragment | ethersUtils.FunctionFragment)[]
+export type ForwardingPathDeclaration = AppOrAddress[]
+
+export type PathOptions = {
+  // The account to sign the transactions with. It is optional
+  // when `actAs` has been set with the connection. If not,
+  // the address has to be passed.
+  actAs?: Address
+
+  // Optionally declare a forwarding path. When not specified,
+  // the shortest path is used instead.
+  path?: ForwardingPathDeclaration
+}
+
+export interface CallScriptAction {
+  to: string
+  data: string
+}
+export interface Annotation {
+  type: string
+  value: any
+}
+export interface PostProcessDescription {
+  description: string
+  annotatedDescription?: Annotation[]
+}
+
+export interface StepDecoded {
```

Comment:
The `StepDecoded` and `StepDescribed` are the two relevant interfaces we are using to construct the `ForwardingPath` description.